packages/matter.js/src/cluster/ClusterServerFactory.ts

```diff
@@ -0,0 +1,122 @@
+            (result as any)[name] = cluster.attributes[name].default
+        }
+    });
+
+    return result;
+}
+
+/**
+ * This factory acts as a central registry for default cluster implementations.
+ */
+export class ClusterServerFactory {
+    private static defaultHandlerProviders = new Map<AnyCluster, () => any>();
+    private static defaultAttributeInitialValues = new Map<AnyCluster, any>;
+
+    /**
+     * Retrieve default values for attributes based on cluster schema.
+     */
+    public static getDefaultAttributeInitialValues<F extends BitSchema, SF extends TypeFromBitSchema<F>, A extends Attributes, C extends Commands, E extends Events>(cluster: Cluster<F, SF, A, C, E>) {
+        const aiv = this.defaultAttributeInitialValues.get(cluster) as AttributeInitialValues<A>;
+        if (aiv == undefined) {
+            return createDefaultAttributeInitialValues(cluster);
+        }
+        return aiv;
+    }
+
+    /**
+     * Register the default implementation for a cluster.
+     * 
+     * @param cluster the cluster to configure
+     * @param defaultHandlerProvider a function that create the cluster handlers
+     * @param defaultAttributeInitialValues optional overrides for default attribute values
+     */
+    public static registerClusterDefaults<F extends BitSchema, SF extends TypeFromBitSchema<F>, A extends Attributes, C extends Commands, E extends Events>(
+        cluster: Cluster<F, SF, A, C, E>, defaultHandlerProvider: () => ClusterServerHandlers<Cluster<F, SF, A, C, E>>,
+        defaultAttributeInitialValues: AttributeInitialValues<A> = createDefaultAttributeInitialValues(cluster)
+    ) {
+        this.defaultHandlerProviders.set(cluster, defaultHandlerProvider);
+        this.defaultAttributeInitialValues.set(cluster, defaultAttributeInitialValues);
+
+        const x = this.defaultAttributeInitialValues.get(cluster);
+        if (x.sceneCount) console.log(x.sceneCount);
+    }
+
+    /**
+     * Create a new cluster server.
+     * 
+     * @param cluster the cluster for the server
+     * @param attributeInitialValues optional overrides for attribute values
+     * @param handlers optional overrides for cluster handlers
+     * @returns 
+     */
+    public static create<F extends BitSchema, SF extends TypeFromBitSchema<F>, A extends Attributes, C extends Commands, E extends Events>(cluster: Cluster<F, SF, A, C, E>, attributeInitialValues?: AttributeInitialValues<A>, handlers?: ClusterServerHandlers<Cluster<F, SF, A, C, E>>) {
+        handlers = Object.assign({}, handlers);
+
+        const provider = this.defaultHandlerProviders.get(cluster);
+        if (provider) {
+            Object.assign(handlers, provider());
+        }
+
+        for (const command in cluster.commands) {
+            if (!(handlers as any)[command]) {
+                (handlers as any)[command] = notImplemented;
+            }
+        }
+
+        // TODO - should we throw an error here if mandatory attributes do not
```

Comment:
Formally yes ... that's why my default clusters always had a full attribute initial values because we can not make sure all mandatory have default values.

Reply:
Yeah I was thinking this was a safety check on the server implementation.

Reply:
With the approach to use default values to populate default we need to check and throw

Reply:
I'll probably add that anyway anyway just as a safety assertion.